src/main/java/org/elasticsearch/cluster/ClusterState.java

```diff
@@ -249,6 +253,10 @@ public ClusterBlocks getBlocks() {
         return this.customs;
     }
 
+    public <T extends Custom> T custom(String type) {
+        return (T) customs.get(type);
```

Comment:
I'm a little fuzzy on this, why do we need to get it out of the `customs` map instead of using the `customPrototype` and getting it there?